src/pgtest/src/lib.rs

```diff
@@ -0,0 +1,333 @@
+// Copyright Materialize, Inc. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! pgtest is a Postgres wire protocol tester using
+//! datadriven test files. It can be used to send [specific
+//! messages](https://www.postgresql.org/docs/current/protocol-message-formats.html)
+//! to any Postgres-compatible server and record received messages.
+//!
+//! The following datadriven directives are supported:
+//! - `send`: Sends input messages to the server. Arguments, if needed,
+//! are specified using JSON. Refer to the associated types to see
+//! supported arguments. Arguments can be omitted to use defaults.
+//! - `until`: Waits until input messages have been received from the
+//! server. Additional messages are accumulated and returned as well.
+//!
+//! Supported `send` types:
+//! - [`Query`](struct.Query.html)
+//! - [`Parse`](struct.Parse.html)
+//! - [`Bind`](struct.Bind.html)
+//! - [`Execute`](struct.Execute.html)
+//! - `Sync`
+//!
+//! For example, to execute a simple prepared statement:
+//! ```pgtest
+//! send
+//! Parse {"query": "SELECT $1::text, 1 + $2::int4"}
+//! Bind {"values": ["blah", "4"]}
+//! Execute
+//! Sync
+//! ----
+//!
+//! until
+//! ReadyForQuery
+//! ----
+//! ParseComplete
+//! BindComplete
+//! DataRow {"fields":["blah","5"]}
+//! CommandComplete {"tag":"SELECT 1"}
+//! ReadyForQuery {"status":"I"}
+//! ```
+
+use anyhow::{anyhow, Result};
```

Comment:
Point of personal preference, no obligation, but I dislike importing `anyhow::Result` or otherwise aliasing the `Result` type. I find that aliasing it makes it harder to see what the error type is for only a small improvement in conciseness. Unfortunately a lot of the ecosystem uses this trick, and our code does too in places, so you're definitely welcome to do this if you prefer, especially in code for which you are the primary maintainer (e.g., this new crate).

The only ask I'd have is that, in other crates, you follow the convention for whatever currently exists. If you wanted to build consensus in one direction or the other and then do a mass conversion one way or another, that's great! But this issue has proven contentious enough that there is definitely a consensus-building phase that'd need to happen first.

Reply:
Thanks for the context. Changed to `anyhow::Result`.